src/julia-syntax.scm

```diff
@@ -332,7 +332,7 @@
 
 ; translate index x from colons to ranges
 (define (expand-index-colon x)
```

Comment:
Looks like this whole function can be removed. We don't allow the `a[i:]` cases anymore, and `:` can just be left as an identifier (it's bound to `Colon()` in Base).